apps/console/src/features/applications/components/modals/application-share-modal.tsx

```diff
@@ -0,0 +1,152 @@
+import { shareApplication } from "../../../organizations/api";
+import { OrganizationInterface } from "../../../organizations/models";
+
+export interface ApplicationShareModalPropsInterface extends ModalProps, IdentifiableComponentInterface {
+    /**
+     * ID of the application to be shared
+     */
+    applicationId: string;
+    /**
+     * Sub Organization list of the current organization.
+     */
+    subOrganizationList: Array<OrganizationInterface>
+}
+
+export const ApplicationShareModal: FunctionComponent<ApplicationShareModalPropsInterface> = (
+    props: ApplicationShareModalPropsInterface
+) => {
+
+    const {
+        applicationId,
+        subOrganizationList,
+        onClose,
+        ["data-componentid"]: componentId,
+        ...rest
+    } = props;
+
+    const dispatch = useDispatch();
+    const currentOrganization = useSelector((state: AppState) => state.organization.organization);
+
+    const [ selectedOrganization, setSelectedOrganization ] = useState<Array<string>>([]);
+
+    const handleShareApplication = useCallback(() => {
+        if (!selectedOrganization) {
+            return;
+        }
+
+        shareApplication(currentOrganization.id, applicationId, selectedOrganization)
+            .then((_response) => {
+                onClose(null, null);
+                dispatch(addAlert({
+                    description: "Application Shared with the organization(s) successfully",
+                    level: AlertLevels.SUCCESS,
+                    message: "Application shared!"
+                }));
+            }).catch((error) => {
+                onClose(null, null);
+                if (error.response.data.message) {
+                    dispatch(addAlert({
+                        description: error.response.data.message,
+                        level: AlertLevels.ERROR,
+                        message: "Application sharing failed!"
+                    }));
+                } else {
+                    dispatch(addAlert({
+                        description: "Application sharing failed. Please try again",
+                        level: AlertLevels.ERROR,
+                        message: "Application sharing failed!"
+                    }));
+                }
+            });
+    }, [ selectedOrganization ]);
+
+    return (
+        <Modal
+            closeOnDimmerClick
+            dimmer="blurring"
+            size="tiny"
+            closeOnDocumentClick={ true }
+            closeOnEscape={ true }
+            data-testid={ `${componentId}-view-certificate-modal` }
+            { ...rest }
+        >
+            <Modal.Header>
+                Share Application
+            </Modal.Header>
+            <Modal.Content>
+                <Form>
+                    <Form.Select
+                        options={ subOrganizationList?.map((subOrganization) => {
+                            return {
+                                key: subOrganization.id,
```

Comment:
Shall we add a null check here? `subjOrganization?.id` etc? Check in other places too

Reply:
This segment is obsolete now since the modal content is revamped